docs/resources/aws_efs_file_system.md

```diff
@@ -0,0 +1,95 @@
+---
+title: About the aws_efs_file_system Resource
+---
+
+# aws\_efs\_file_system
+
+Use the `aws_efs_file_system` InSpec audit resource to test the properties of a single AWS EFS file system.
+
+## Syntax
+
+An `aws_efs_file_system` resource block declares the tests for a single AWS EFS file system by either file system id or creation token.
+
+    describe aws_efs_file_system('fs-12345678') do
+      it                         { should be_encrypted }
+      its('size_in_bytes.value') { should cmp 6144 }
+    end
+
+    describe aws_efs_file_system(creation_token: 'my-token') do
+      its('encrypted')       { should cmp true }
+      its('throughput_mode') { should eq 'bursting' }
+    end
+
+#### Parameters
+
+One of either the EFS file system id or creation token must be provided.
+
+##### file\_system\_id _(required if `creation_token` not provided)_
+
+The ID of the EFS file system. This is in the format of `fs-` followed by 8 or 17 hexadecimal characters.
+This can be passed either as a string or as a `file_system_id: 'value'` key-value entry in a hash.
+
+##### creation_token _(required if `file_system_id` not provided)_
```

Comment:
I would wonder whether this is something that the end-user would typically know. Is this available through the UI after the resource has been created?
It might be nice to be able to filter using the name field if that's an option. (not sure if that is an optional field?) https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EFS/Client.html#describe_file_systems-instance_method

Reply:
The `file_system_id` is available through UI or CLI after the resource has been created.

The `name` is available if the `Name` tag has been defined at creation.
I will add that as a filtering option. 👍

Reply:
Is `creation_token` available through the UI though?

Reply:
Correction to my previous comment:
- `name`: This can not be used to query the file systems. However, its value can be accessed by `a_file_system.name` as opposed to other tags.
- Only `creation_token` and `file_system_id` can be used to query a specific file system.
- `creation_token` can be provided at resource creation and it is available through UI/CLI. If not provided it will be auto-assigned.
https://docs.aws.amazon.com/cli/latest/reference/efs/describe-file-systems.html

Reply:
Cool thanks for the clarification on that.
Would be nice to add `name` to the plural resource filter table

Reply:
Added `names` to the plural resource.
Thanks.